modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/GossipDaemon.scala

```diff
@@ -40,115 +39,127 @@ object GossipDaemon {
+          batch.filter { case (_, s) => s.value.isInstanceOf[CommonRumorBinary] } ++
+            batch.filter { case (_, s) => s.value.isInstanceOf[PeerRumorBinary] }
+              .sortBy(_._2.asInstanceOf[Signed[PeerRumorBinary]].ordinal)
+
+        private def handleRumor(har: HashAndRumor): F[Unit] = har match {
+          case (hash, signedRumor) =>
+            rumorHandler
+              .run((signedRumor.value, nodeId))
+              .semiflatTap(_ => metrics.updateRumorsConsumed("success", signedRumor))
+              .getOrElseF {
+                logger.debug(s"Unhandled rumor ${signedRumor.value.show} with hash ${hash.show}") >>
+                  metrics.updateRumorsConsumed("unhandled", signedRumor)
+              }
+              .handleErrorWith { err =>
+                logger.error(err)(s"Error handling rumor ${signedRumor.value.show} with hash ${hash.show}") >>
+                  metrics.updateRumorsConsumed("error", signedRumor)
+              }
         }
-      } yield ()
-
-    private def runGossipRound(activeHashes: List[Hash], seenHashes: List[Hash], peer: Peer): F[Unit] =
-      for {
-        startRequest <- StartGossipRoundRequest(activeHashes).pure[F]
-        startResponse <- gossipClient.startGossiping(startRequest).run(peer)
-        inquiry = startResponse.offer.diff(seenHashes)
-        answer <- rumorStorage.getRumors(startResponse.inquiry)
-        endRequest = EndGossipRoundRequest(answer, inquiry)
-        _ <- metrics.updateRumorsSent(answer)
-        endResponse <- gossipClient.endGossiping(endRequest).run(peer)
-        _ <- rumorQueue.offer(endResponse.answer)
-        _ <- metrics.updateRumorsReceived(endResponse.answer)
-      } yield ()
+
+        private def selectPeers: F[Unit] =
+          for {
+            _ <- Temporal[F].sleep(cfg.interval)
+            peers <- clusterStorage.getPeers
+            selectedPeers <- Random[F].shuffleList(peers.toList).map(_.take(cfg.fanout))
+            _ <- selectedPeers.traverse(selectedPeerQueue.offer)
+          } yield ()
+
+        private def runGossipRounds: F[Unit] =
+          Stream
+            .fromQueueUnterminated(selectedPeerQueue)
+            .parEvalMapUnordered(cfg.maxConcurrentRounds) { peer =>
+              Temporal[F]
+                .timeout(runGossipRound(peer), cfg.roundTimeout)
+                .flatTap(_ => metrics.incrementGossipRoundSucceeded)
+                .handleErrorWith { err =>
+                  logger.error(err)(s"Error running gossip round with peer ${peer.show}") >>
+                    Spawn[F].start(healthcheck.triggerCheckForPeer(peer)).void
+                }
+            }
+            .compile
+            .drain
+
+        private def runGossipRound(peer: Peer): F[Unit] =
+          for {
+            activeHashes <- rumorStorage.getActiveHashes
+            seenHashes <- rumorStorage.getSeenHashes
+            startRequest <- StartGossipRoundRequest(activeHashes).pure[F]
```

Comment:
I just noticed that this `.pure[F]` can be removed (before this was the first expression in the for-comprehension so it was actually used).